pkg/network/setup/filters_test.go

```diff
@@ -47,4 +49,141 @@ var _ = Describe("Network setup filters", func() {
 			Expect(network.FilterNetsForVMStartup(vmi)).To(Equal([]v1.Network{*v1.DefaultPodNetwork()}))
 		})
 	})
+
+	Context("FilterNetsForLiveNICChange", func() {
+		const (
+			net1Name = "net1"
+			net2Name = "net2"
+			nad1Name = "nad1"
+			nad2Name = "nad2"
+		)
+
+		multusAndDomainInfoSource := vmispec.NewInfoSource(vmispec.InfoSourceMultusStatus, vmispec.InfoSourceDomain)
+
+		It("Should return an empty list when there are no networks to hot plug/unplug", func() {
+			vmi := libvmi.New(
+				libvmi.WithInterface(*v1.DefaultBridgeNetworkInterface()),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(net1Name)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(net2Name)),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithNetwork(libvmi.MultusNetwork(net1Name, nad1Name)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(net2Name, nad2Name)),
+				libvmistatus.WithStatus(
+					libvmistatus.New(
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       "default",
+							InfoSource: vmispec.InfoSourceDomainAndGA,
+						}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       net1Name,
+							InfoSource: multusAndDomainInfoSource,
+						}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       net2Name,
+							InfoSource: multusAndDomainInfoSource,
+						}),
+					),
+				),
+			)
+
+			Expect(network.FilterNetsForLiveNICChange(vmi)).To(BeEmpty())
+		})
+
+		It("Should return a network to hotplug", func() {
```

Comment:
Please consider adding more context, e.g.: "when the requested interface exist in pod but not attached to the guest"

Reply:
Done.